backend/service/k8s/job.go

```diff
@@ -20,6 +22,31 @@ func (s *svc) DeleteJob(ctx context.Context, clientset, cluster, namespace, name
 	return cs.BatchV1().Jobs(cs.Namespace()).Delete(ctx, name, opts)
 }
 
+func (s *svc) CreateJob(ctx context.Context, clientset, cluster, namespace string, jobConfig *structpb.Value) (*k8sapiv1.Job, error) {
+	cs, err := s.manager.GetK8sClientset(ctx, clientset, cluster, namespace)
+	if err != nil {
+		return nil, err
+	}
+
+	// convert Job config into a runtime object and then type cast to a
+	// batch job
+	config := []byte(jobConfig.GetStringValue())
+	decode := scheme.Codecs.UniversalDeserializer().Decode
+	obj, _, err := decode(config, nil, nil)
+	if err != nil {
+		return nil, err
+	}
+	job := obj.(*v1.Job)
```

Comment:
i'd think i'd actually push this conversion code up to the module since it deals with user input transformation. if other code paths want to create a job, they wouldn't need to go back and forth between yaml and the go struct.

Reply:
Ah ok and then change the function signature of this method to take in the K8s job directly?

Reply:
yep!

Reply:
Done